### ISSUE TITLE
Reproduce "test" accuracy

### ISSUE THREAD

**Issue by tbachlechner** [akamaster/pytorch_resnet_cifar10#18]
I'm having trouble reproducing the test accuracy that you quote in the Readme. In particular running your precise code for ResNet56 I get the best validation error rate of 7.36(16)%. This differs from your quoted value of 6.61% by 5 sigma. How exactly did you determine the quoted test accuracy of your model?

**Comment by akamaster** [comment]
The pretrained models with reported errors are released along with the code. For example, here is the pretrained model for [ResNet56](https://github.com/akamaster/pytorch_resnet_cifar10/blob/master/pretrained_models/resnet56-4bfd9763.th), you can verify its accuracy is as reported. 

The reported accuracies were obtained by running the training script, no bells and whistles. If during your training you have reached a different accuracy, I would suggest retraining, as the SGD is a stochastic algorithm, thus results can differ from run-to-run.

**Comment by tbachlechner** [comment]
Thanks, I ran it 6 times and for each run I find the following best (top 1) errors:
`7.4, 7.29, 7.4, 6.89, 8.42, 7.95`
You can probably run it many times and find a lower test error by statistical fluctuation, but I'm not sure how well that characterizes the model since it depends on the number of runs. I ask because in https://arxiv.org/pdf/2003.04887.pdf we're reporting an error of  7.37 +/- 0.06 % for this exact implementation to compare to our modification, so I wanted to understand the source of the discrepancy.

**Comment by akamaster** [comment]
That is strange. I've had fluctuations above 7 but never so many. I'll re-run training to see what's happening. In meantime, can you share the training log? Also, are you running the training for 200 epochs?

**Comment by tbachlechner** [comment]
Thanks! Let me know what you find. Here is a link to the log for three ResNet56 runs, training up to 200 epochs: https://www.dropbox.com/s/uuzmrfxtj0u64ha/log_resnet56.txt?dl=0

**Comment by akamaster** [comment]
what is the weight decay? momentum? also, the pytorch version?

**Comment by tbachlechner** [comment]
I'm using all your default parameters, i.e. `batch_size = 128, momentum = 0.9, weight_decay = 1e-4`.
I use Python 3.7.6 and PyTorch 1.4.0.

**Comment by akamaster** [comment]
I'm running on PyTorch 1.2.0, but I don't think this is a problem. While I'm re-running the resnet56, I would suggest couple of things:
- re-download the data(CIFAR10), might be some images got corrupted?
- can you also makre sure that pre-trained resnet56 from the repo achives the reported error?

**Comment by tbachlechner** [comment]
Thanks, I loaded the parameters from the model you provided online and successfully reproduced the accuracy of 6.61% that you quote.

```
Test: [0/79]	Time 0.270 (0.270)	Loss 0.2958 (0.2958)	Prec@1 96.094 (96.094)
Test: [50/79]	Time 0.027 (0.032)	Loss 0.2139 (0.3548)	Prec@1 96.094 (93.244)
 * Prec@1 93.390
```

I can re-download the training data and run it again.

**Comment by akamaster** [comment]
The training finished with an error of 7.5% on PyTorch 1.2, so there is definitely something going on. Then I re-run on PyTorch 1.1 and got 6.7%. Currently, I'm not sure whether this is PyTorch's fault or just the variability of the training itself. However, I don't think I will look into this issue, so I would recommend using older PyTorch versions for now.
Edit: 
here is the log of training with 7.5% error on PyTorch 1.2, https://pastebin.com/MwQpdRxw
here is the log of training with 6.7% error on PyTorch 1.1, https://pastebin.com/s7V1Pudt

Note: I have not run multiple experiments to compute the mean and std of the error.

**Comment by tbachlechner** [comment]
Ok, great, thanks for checking that out. At least now I know the source of the discrepancy and we'll report our PyTorch version along with the accuracy we find. Beautiful implementation by the way! You've made my life very easy.
